README.md

```diff
@@ -45,45 +45,76 @@ TBD, see the Hawk spec for inspiration
 
 Production services implementing this spec must only accept requests using HTTPS.
 
-## Overview of Header and Signature
+## Overview of Request and Response Headers
 
-The pseudocode below illustrates construction of the HTTP "Authorization" header and signature:
+
+### Client
+The HTTP client signs the request by adding the following HTTP headers:
 
 ```
-Authorization: acquia-http-hmac realm="Example",
-               id="identifier",
-               nonce="d1954337-5319-4821-8427-115542e08d10",
-               version="2.0",
-               headers="",
-               signature="Signature"
+Authorization : <HMACAuthorization>
+X-Authorization-Timestamp : <Unix Timestamp In Seconds>
+X-Authorization-Content-SHA256 : <HashedContent> (if Content-Length > 0)
+```
+The HTTP client must not add the following HTTP Header to the request:
 
-X-Authorization-Timestamp: 1432075982
+X-Authenticated-Id (reserved for internal server to server communication)
 
-Signature = Base64( HMAC( SecretKey, Signature-Base-String ) );
+### Server
 
-Signature-Base-String =
-    HTTP-Verb + "\n" +
-    Host + "\n" +
-    Path + "\n" +
-    Query-Parameters + "\n" +
-    Authorization-Header-Parameters + "\n" +
-    for each added signed header: added-signed-header-name:added-signed-header-value + "\n" +
-    Timestamp +
-    (omit below if Content-Length is 0)
-    "\n" + Content-Type +
-    "\n" + Body-Hash
-;
+If the server authenticates the request successfully, it will add the following HTTP headers to the response (for all non-HEAD requests):
+
+```
+X-Server-Authorization-HMAC-SHA256: <HMACServerAuthorization>
 ```
 
-`"\n"` denotes a Unix-style line feed (ASCII code `0x0A`).
+The client should verify the response HMAC which authenticates the response body back from the server.
 
-#### Secret Key
+## Overview of Request Header and Signature
 
-The secret key should be a 256 to 512 bit binary value. The secret key will normally be stored as a base64-encoded or hex-encoded string representation, but must be decoded to the binary value before use.
+The pseudocode below illustrates construction of the HTTP "Authorization" header and signature:
 
-#### Authorization Header
+```
+HMACAuthorization = "acquia-http-hmac" + " " +
+                "realm=" + Provider + "," +
```

Comment:
It might be nice to mention in the pseudo code that each attribute value is urlencoded. Up to you obviously.